pages/compare.py

```diff
@@ -242,143 +211,23 @@ def create_prompt_model(model_id, prompt, position):
-                          ),
-                      ),
-                      node_inputs=[resources_pb2.NodeInput(node_id="prompt",)],
-                  ),
-              ],
-          ),
-      ],
-  )
-
-  # FIXME(zeiler): i think that if the user is logged in then postmodeloutputs or
-  # postworkflowresults will fail because these models/workflows are not made publicly visible.
-  # When the are anonymous then we fall back to the PAT provided in the secrets file.
-  response = secrets_stub.PostWorkflows(req)
-  if response.status.code != status_code_pb2.SUCCESS:
-    show_error("PostWorkflows", response)
-
-  if DEBUG:
-    st.json(json_format.MessageToDict(response, preserving_proto_field_name=True))
-
-  return response.workflows[0]
-
-
-def delete_workflow(workflow):
-  response = secrets_stub.DeleteWorkflows(
-      service_pb2.DeleteWorkflowsRequest(
-          user_app_id=userDataObject,
-          ids=[workflow.id],
-      ))
-  if response.status.code != status_code_pb2.SUCCESS:
-    show_error("DeleteWorkflows", response)
-  else:
-    print(f"Workflow {workflow.id} deleted")
-
-
 @st.cache_resource
-def run_workflow(input_text, workflow):
-  response = user_or_secrets_stub.PostWorkflowResults(
-      service_pb2.PostWorkflowResultsRequest(
-          user_app_id=userDataObject,
-          workflow_id=workflow.id,
-          inputs=[
-              resources_pb2.Input(
-                  data=resources_pb2.Data(text=resources_pb2.Text(raw=input_text,),),),
-          ],
-      ))
-  if response.status.code != status_code_pb2.SUCCESS:
-    show_error("PostWorkflowResults", response)
-
-  if DEBUG:
-    st.json(json_format.MessageToDict(response, preserving_proto_field_name=True))
-
-  return response
-
 
-@st.cache_resource
+#Used predict_by_bytes new SDK function.
 def run_model(input_text, model):
 
   m = API_INFO[model]
-  start_time = time.time()
+  model_obj=Model(model_id=m["model_id"], user_id=m["user_id"], app_id=m["app_id"])
   while True:
+    try:
+      response = model_obj.predict_by_bytes(bytes(
+      input_text, 'utf-8'),
+      "text")
 
-    response = user_or_secrets_stub.PostModelOutputs(
-        service_pb2.PostModelOutputsRequest(
-            user_app_id=resources_pb2.UserAppIDSet(user_id=m['user_id'], app_id=m['app_id']),
-            model_id=m['model_id'],
-            inputs=[
-                resources_pb2.Input(
-                    data=resources_pb2.Data(text=resources_pb2.Text(raw=input_text,),),),
-            ],
-        ))
-
-    if response.outputs and response.outputs[0].status.code == status_code_pb2.MODEL_DEPLOYING and time.time(
-    ) - start_time < 60 * 10:
-      st.info(f"{model.split(':')[0]} model is still deploying, please wait...")
-      time.sleep(5)
-      continue
-
-    if response.status.code != status_code_pb2.SUCCESS:
-      show_error("PostModelOutputs", response)
-    else:
-      break
+    except Exception as e:
+      st.write(f"Error: {e}")
```

Comment:
We could use the `show_error` function incase of any errors

Reply:
modified above and also changed post_input function as well to new SDK